lib/ffmpeg.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var childProcess = require('child_process');
+var spawnSync = childProcess.spawnSync;
+var spawn = childProcess.spawn;
+
+var pluginErrors = require('./errors');
+
+var probeParams = ['-v','quiet', '-print_format', 'json', '-show_format'];
+
+var preFlightCheck = spawnSync('ffprobe', ['-version']).error === undefined;
+
+module.exports = {
+
+  preFlightCheck: function() {
+    if (!preFlightCheck) {
+      throw new Error(pluginErrors.preflightError);
+    }
+  },
+
+  getFileInfo: function(filePath) {
+    return new Promise(function(resolve, reject) {
+      var ffprobe = spawn(
+        'ffprobe',
+        probeParams.concat([filePath])
+      );
+      var fileInfo = '';
+
+      ffprobe.stdout.on('data', function(response) {
+        fileInfo += response.toString();
+      });
+
+      ffprobe.stdout.on('close', function() {
+        var infoObject = JSON.parse(fileInfo);
```

Comment:
Turn this function into `resolve(fileInfo);`. This parsing should be done in the promise chain, it'll make new Promise's body simpler and will also allow us to catch JSON.parse errors in the Promise chain. Otherwise if fileInfo is invalid JSON, the build will just stall here.